idseq_pipeline/commands/common.py

```diff
@@ -123,14 +123,14 @@ def execute_command(command, progress_file=None):
 
 
 def remote_command(base_command, key_path, remote_username, instance_ip):
-    return 'ssh -o "StrictHostKeyChecking no" -i %s %s@%s "%s"' % (key_path, remote_username, instance_ip, base_command)
+    return 'ssh -o "StrictHostKeyChecking no" -o "ConnectTimeout 15" -i %s %s@%s "%s"' % (key_path, remote_username, instance_ip, base_command)
```

Comment:
Is it going to exit gracefully or with an error if timeout is exceeded?

Reply:
With an error, which would in turn cause the execute_command function to raise an exception.

The default TCP timeout seems to be between 3-4 minutes, and if a machine goes away, this causes the polling for SSH servers to take 3-4 minutes each time that dead machine is polled.   It's better to err out quickly in those cases.  And in general, I've always found it necessary to shorten this timeout in past.

Reply:
Note this is a timeout just for the handshake to open the TCP connection, not for the command that runs within the SSH session.   There is no timeout for that command.